crates/runtime/src/instance/allocator/pooling.rs

```diff
@@ -64,6 +125,49 @@ fn round_up_to_pow2(n: usize, to: usize) -> usize {
     (n + to - 1) & !(to - 1)
 }
 
+/// The "backend", or implementation strategy, to be used in providing
+/// virtual memory for the pooling allocator.
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+pub enum PoolingBackend {
+    /// Use conventional POSIX memory-mapping syscalls, `mmap()` and
+    /// `mprotect()`. Should work on any Unix-like system.
+    Mmap,
+    /// Use userfaultfd. Supported only on Linux.
+    #[cfg(feature = "uffd")]
+    Uffd,
+    /// Use memfd. Supported only on Linux.
+    #[cfg(feature = "memfd-allocator")]
+    MemFd,
+}
+
+impl std::default::Default for PoolingBackend {
+    fn default() -> Self {
+        cfg_if::cfg_if! {
+            // This may look a bit complex, but the intent is to allow
+            // for testability of dynamic selection of non-default
+            // backends.
+            //
+            // The basic logic is: if `default-*-backend` is set, then
+            // that is the default backend, as long as it is
+            // available. Otherwise, the default is the first
+            // available of: uffd, memfd, mmap.
+            if #[cfg(feature = "default-mmap-backend")] {
```

Comment:
Personally I don't really feel the need to have separate features for default backends. If this is purely for test coverage then we could have normal `cargo test` just loop through the backends or otherwise just pick a random one for each test so we get more coverage over time or something like that.